lib/puppetserver/ca/puppet_config.rb

```diff
@@ -72,17 +72,33 @@ def resolve_settings(overrides = {})
         ssldir = overrides.fetch(:ssldir, '$confdir/ssl')
         settings[:ssldir] = substitutions['$ssldir'] = ssldir.sub('$confdir', confdir)
 
+        certdir = overrides.fetch(:certdir, '$ssldir/certs')
+        settings[:certdir] = substitutions['$certdir'] = certdir.sub(unresolved_setting, substitutions)
+
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
+        # Note the call to hostname below, this is horrific....
+        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, `hostname`.chomp)
+        server = overrides.fetch(:server, '$certname')
+        settings[:server] = substitutions['$server'] = server.sub(unresolved_setting, substitutions)
+
+        settings[:ca_server] = overrides.fetch(:ca_server, '$server').sub(unresolved_setting, substitutions)
```

Comment:
Do we need the ```.sub(unresolved_setting, substitutions)``` here too?  maybe it could just get evaluated below with the ```.gsub``` ?

Reply:
Good catch